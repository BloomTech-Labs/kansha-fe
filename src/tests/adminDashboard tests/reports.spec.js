import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ReportsPage from '../../components/AdminDashboard/AdminDashboard';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { renderWithRouter } from '../testUtils';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

let ReportsComponent = (
	<BrowserRouter>
		<ReportsPage />
	</BrowserRouter>
);
afterEach(cleanup);

describe('Reports', () => {
	const mockStore = configureStore();
	let store, wrapper;

	it('renders without crashing', () => {
		store = mockStore();

		render(<Provider store={store}>ReportsComponent</Provider>);
	});

	it('matches snapshot', () => {
		const tree = renderer.create(
			<Provider store={store}>ReportsComponent</Provider>,
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('it displays header text', () => {
		const { getByText } = render(
			<Provider store={store}>ReportsComponent</Provider>,
		);
		getByText(/Overview/i);
		getByText(/Recognition/i);
		getByText(/Participation/i);
		getByText(/Most Thanked/i);
		getByText(/Most Thankful/i);
	});
});
