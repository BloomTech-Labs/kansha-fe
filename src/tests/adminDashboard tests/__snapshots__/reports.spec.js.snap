// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports matches snapshot 1`] = `"ReportsComponent"`;
